This Week in Matrix July 24 (#3551)
matrix-org/matrix.org: content/blog/2026/07/2026-07-24-twim.md

> ![Matrix Conf Merchandise preview](https://matrix.org/blog/img/2026-07-24-merch-twim.jpg)

Reviewer: hmm @MTRNord wdyt? i'm guessing internal links wouldn't be supported.

```suggestion
![Matrix Conf Merchandise preview](/blog/img/2026-07-24-merch-twim.jpg)
```

Reviewer 2: iirc the real internal links only work for stuff in the `content` folder. So yeah this has to be like this without the @ since we have images not near the blogposts

>
> #### More Volunteering Slots Available
>
> The Matrix Conference is made possible thanks to our awesome sponsors financing it just as well as our awesome volunteer team staffing the welcome desk, operating video equipment, and generally jumping in whenever an extra hand is needed. We have onboarded a lot of volunteers already and our shift booking system was totally booked out while we worked through all these applications from you amazing people, and some have already asked when more will be available. This is (as of writing) the case! If you were waiting for one to free up again, head to our [volunteering system](https://conference.matrix.org/volunteer/) and sign up for two shifts to receive a free ticket to the Conference, the limited edition team-only zipper hoodie in special colours, and of course a lot of experience, satisfaction and fun from helping making The Matrix Conference happen!
>
> If you've already signed up a while ago and have received neither an email nor invite on Matrix from us yet, please reach out to us over at #events-wg:matrix.org.
>

> > See you soon, your [Events WG](https://matrix.org/foundation/working-groups/events/).

Reviewer: ```suggestion
> See you soon, your [Events WG](@/foundation/working-groups/events.md).
```

Reviewer: hm, this is marked as resolved, but when I check out the branch it isn't.

## Dept of Working Groups 💪

### Homeserver Decentralisation Working Group ([website](@/foundation/working-groups/homeserver-decentralisation.md))

The Homeserver Decentralisation Working Group aims to promote decentralisation of the Matrix network by fostering a smoother onboarding experience (to homeservers and clients) on homeservers other than matrix.org, and might propose additional measures in the ecosystem as a whole.

[Nicolas Da Mutten](https://matrix.to/#/@nicolas:damutten.ch) announces

> [@this-week-in:matrix.org](https://matrix.to/#/@this-week-in:matrix.org)

> >

Reviewer 2: ```suggestion

```

>
> Lots of updates in spec world this week! New MSCs, MSCs being closed in favour of others or due to lack of time. Don't forget that a closed MSC can always be opened under a new MSC # if someone thinks its worth pursuing - the idea isn't dead forever!
>
> [MSC4319](https://github.com/matrix-org/matrix-spec-proposals/pull/4319) entered final comment period this week. It proposes that additional `m.room.member` events be included when receiving room state during an invite or knock. This allows for clients (and users) to receive information about who invited them (or knocked on the room), when the invite (or knock) occurred etc.
>
> FCP ends in 3 days, so go and check the MSC out for any last minute additions/changes if this is an area you're interested in!

## Dept of Clients 📱

### Nexus ([website](https://nexus.federated.nexus))

Matrix client made with Flutter and a Gomuks backend.

[QuadRadical (Ping)](https://matrix.to/#/@quadradical:federated.nexus) announces

> #### What's new?
> Hello TWIM! You may have been wondering what has been happening with Nexus, as there hasn't been an update in a while. Well, here it is:
>
> - Support for OAuth login, and removal of UIAA login

> >   (please insert here [https://git.federated.nexus/Nexus/nexus/raw/branch/main/assets/twim/oauth.webp](https://git.federated.nexus/Nexus/nexus/raw/branch/main/assets/twim/oauth.webp))

Reviewer: did you spot this? my last understanding is that we would prefer TWIMs to be selfcontained incl media, so that if the src ever disappears twim remains intact. that means we would reupload and possibly [recompress](https://github.com/matrix-org/matrix.org/blob/205e9cd43722e2888ddb2cccf0c8a2af2d193960/CONTENT.md?plain=1#L96-L97) this - retaining sufficient quality of course.

Author: oh! I missed that. thanks. Will grab.

Author: copied webp over. I've been using PNG for OG images, maybe I should use webp.

Reviewer 2: @andypiper For OG we should use PNG. Some social networks seem to not support modern image formats sadly. PNG in the past had the best coverage.

### Element X iOS ([website](https://github.com/vector-im/element-x-ios))

A total rewrite of Element iOS using the Matrix Rust SDK underneath and targeting devices running iOS 17+.

[Doug](https://matrix.to/#/@douge:matrix.org) says

> This week we’ve been hard at work with the 3 next features for the app:
>
> - We saw our first data syncing to the app from Synapse for User Status 🎉
> - We’re adding support for Media Galleries to match Element X Android. Sending is merged and we’re now iterating on the timeline rendering. 🖼️
> - Work on Message Search has resumed after some team holidays and is looking really good! 🔍
>
> Other than that 26.07.4 landed on the App Store on Monday with all the changes mentioned last week.

### Element X Android ([website](https://github.com/vector-im/element-x-android))

Android Matrix messenger application using the Matrix Rust SDK and Jetpack Compose.

[Jorge](https://matrix.to/#/@jorgem:element.io) says

> This has been a quiet week, mostly focused on continuing the work on WIP features like user status, and fixing bugs and accessibility issues.

> >
> > [Element X Android v26.07.1](https://github.com/element-hq/element-x-android/releases/tag/v26.07.1) has been published and it contains all the changes we told you about in previous TWIM messages.
> 
> ## Dept of SDKs and Frameworks 🧰
> 
> [OdoItal](https://matrix.to/#/@me:shahpaarth.com) reports
> 
> > [matrix-nio](https://github.com/matrix-nio/) 0.26.0
> >
> > This release swaps the end-to-end encryption backend from libolm to [vodozemac](https://github.com/matrix-nio/vodozemac-python), drops end-of-life Python versions, and modernizes the project's tooling.

Reviewer 2: ```suggestion

## Dept of SDKs and Frameworks 🧰

### matrix-nio

[OdoItal](https://matrix.to/#/@me:shahpaarth.com) reports

> **matrix-nio 0.26.0!**
> This release swaps the end-to-end encryption backend from libolm to [vodozemac](https://github.com/matrix-nio/vodozemac-python), drops end-of-life Python versions, and modernizes the project's tooling.
```

>
> **This is a breaking release.** Encryption now runs on vodozemac instead of libolm/`python-olm`, some encryption defaults have changed, and Python 3.9 and older are no longer supported.
>

> > ## ⚠️ Breaking changes

Reviewer 2: ```suggestion
> #### ⚠️ Breaking changes
```

heading levels for all of matrix-nio are off by two :)

>
> - **Encryption backend replaced: `libolm`/`python-olm` → `vodozemac`.** The `e2e` extra now depends on [`vodozemac`](https://pypi.org/project/vodozemac/), and `libolm` is no longer required as a system dependency.
>     - **Store migration is automatic** on first load. Stores created with a `libolm` pickle version older than `4` (roughly, pre-December 2021) can only be migrated if the optional `python-olm >= 3.2.7` package is installed at upgrade time — otherwise those pickles cannot be read.
>     - **`hmac-sha256` is gone for SAS verification.** Only `hkdf-hmac-sha256` is offered as a message authentication code, so verifying against clients that only support `hmac-sha256` is no longer possible.
>     - **`Account.remove_one_time_keys()` has been removed**, as it is no longer supported by the underlying library.
> - **Dropped Python 3.8 and 3.9.** The minimum supported version is now **Python 3.10**. **Python 3.14 is now supported.** Supported: 3.10, 3.11, 3.12, 3.13, 3.14.
>

> > ## Additions

Reviewer 2: ```suggestion
> #### Additions
```

>
> - Added `unread_thread_notifications` to `SyncResponse`.
>

> > ## Bug fixes

Reviewer 2: ```suggestion
> #### Bug fixes
```

>
> - Fixed `get_openid_token`, which requires an empty JSON body to be sent.
> - Fixed printing of `FileResponse` when a download is saved to a file.
>

> > ## Internal / maintenance

Reviewer 2: ```suggestion
> #### Internal / maintenance
```

>
> - Adopted [`uv`](https://docs.astral.sh/uv/) for project management, applied `pyupgrade` and linting, and bumped dependencies.
> - Unpinned dependencies for more flexible version resolution.
> - Upgraded the underlying `vodozemac` to **0.10.0**.

### vodozemac ([website](https://github.com/matrix-org/vodozemac))

An implementation of Olm and Megolm in pure Rust.

[OdoItal](https://matrix.to/#/@me:shahpaarth.com) says

> [vodozemac-python](https://github.com/matrix-nio/vodozemac-python) 0.10.0
>
> This release upgrades the underlying [vodozemac](https://github.com/matrix-org/vodozemac/releases#release-0.10.0) Rust library to **0.10.0**. The Python package version tracks the Rust crate, so this is `0.10.0`.
>
> **This is a breaking release.** The public Python API is nearly unchanged, but several operations that previously always succeeded can now raise an exception, and Python 3.9 is no longer supported.
>

> > ## ⚠️ Breaking changes

Reviewer 2: ```suggestion
> #### ⚠️ Breaking changes
```

>
> - **Some operations are now fallible.** As of upstream vodozemac 0.10.0, Diffie-Hellman can reject malicious/low-order keys, which makes the following methods able to raise where they previously always returned a value:
>     - `Account.create_outbound_session` → may raise `SessionCreationException`
>     - `Session.encrypt` → may raise the new `OlmEncryptionException`
>     - `PkEncryption.encrypt` → may raise `PkDecodeException`
>   Method signatures are unchanged; wrap these calls in error handling if you process untrusted keys.
> - **Stricter Ed25519 signature verification is now mandatory.** Non-strict verification has been removed upstream. Signatures that were maliciously crafted and previously accepted may now be rejected.
> - **Dropped Python 3.9.** The minimum supported version is now **Python 3.10**. **Python 3.14 is now supported.** Supported: 3.10, 3.11, 3.12, 3.13, 3.14.
>

> > ## Additions

Reviewer 2: ```suggestion
> #### Additions
```

>
> - New `OlmEncryptionException`, raised on Olm encryption failures (mirrors the existing `OlmDecryptionException`).
>

> > ## Internal / maintenance

Reviewer 2: ```suggestion
> #### Internal / maintenance
```

>
> - Upgraded `pyo3` 0.28 → **0.29** and `maturin` to **1.14**.
> - Bumped `thiserror` and refreshed transitive Rust and Python dev dependencies.
> - Session configuration continues to use protocol **version 1** for all Olm/Megolm sessions; this is unchanged from 0.9.

## Dept of Ops 🛠

### Matrix Connectivity Tester ([website](https://connectivity-tester.mtrnord.blog))

A web interface to debug and troubleshoot Matrix connectivity issues.

[MTRNord (they/them)](https://matrix.to/#/@mtrnord:mtrnord.blog) reports

> ### Connectivity Tester

> >

Reviewer 2: ```suggestion
```

Duplicate header :)

+++
date = "2026-07-24T14:16:46+00:00"
title = "This Week in Matrix 2026-07-24"
path = "/blog/2026/07/24/this-week-in-matrix-2026-07-24"

[taxonomies]
author = ["Andy Piper"]
category = ["This Week in Matrix"]

[extra]
image = "https://matrix.org/blog/img/2026-07-24-twim.png"
+++

## Matrix Live S13E03 – Homeserver Decentralisation Working Group

{{youtube_player(video_id="eFxOelR-TEc")}}

## Dept of Events and Talks 🗣️

[HarHarLinks](https://matrix.to/#/@kim:sosnowkadub.de) announces

> ### The Matrix Conference

> >

Reviewer: ```suggestion
### The Matrix Conference

[HarHarLinks](https://matrix.to/#/@kim:sosnowkadub.de) announces

```

>

> > ![Matrix Conf Merchandise preview](https://matrix.org/blog/img/2026-07-24-merch-twim.jpg)

Reviewer: this will not work (in the preview) as long as it keeps saying matrix.org

```suggestion
> ![Matrix Conf Merchandise preview](/blog/img/2026-07-24-merch-twim.jpg)
```

We also like actual accessibility of our content. How detailed and reasonable a description to write though is up to you; this one is quite abstract.

>
> **Reserve yours at <https://conference.matrix.org/register/>!**
>
> PS: You will need to make sure your preorder is picked up during the Conference in Malmö, but you can totally order just your merch today if you're not yet sure which ticket you are going to get!
>
> #### More Volunteering Slots Available
>
> The Matrix Conference is made possible thanks to our awesome sponsors financing it just as well as our awesome volunteer team staffing the welcome desk, operating video equipment, and generally jumping in whenever an extra hand is needed. We have onboarded a lot of volunteers already and our shift booking system was totally booked out while we worked through all these applications from you amazing people, and some have already asked when more will be available. This is (as of writing) the case! If you were waiting for one to free up again, head to our [volunteering system](https://conference.matrix.org/volunteer/) and sign up for two shifts to receive a free ticket to the Conference, the limited edition team-only zipper hoodie in special colours, and of course a lot of experience, satisfaction and fun from helping making The Matrix Conference happen!
>

> > If you've already signed up a while ago and have received neither an email nor invite on Matrix from us yet, please reach out to us over at #events-wg:matrix.org.

Reviewer 2: ```suggestion
> If you've already signed up a while ago and have received neither an email nor invite on Matrix from us yet, please reach out to us over at [#events-wg:matrix.org](https://matrix.to/#/#events-wg:matrix.org).
```

>
> [MSC4319](https://github.com/matrix-org/matrix-spec-proposals/pull/4319) entered final comment period this week. It proposes that additional `m.room.member` events be included when receiving room state during an invite or knock. This allows for clients (and users) to receive information about who invited them (or knocked on the room), when the invite (or knock) occurred etc.
>
> FCP ends in 3 days, so go and check the MSC out for any last minute additions/changes if this is an area you're interested in!

## Dept of Clients 📱

### Nexus ([website](https://nexus.federated.nexus))

Matrix client made with Flutter and a Gomuks backend.

[QuadRadical (Ping)](https://matrix.to/#/@quadradical:federated.nexus) announces

> #### What's new?
> Hello TWIM! You may have been wondering what has been happening with Nexus, as there hasn't been an update in a while. Well, here it is:
>
> - Support for OAuth login, and removal of UIAA login
>
> ![Nexus's OAuth login screen](/blog/img/2026-07-24-nexus-oauth.webp)
>
> - Support for sending media (images, videos, files, etc)

> > - Support for viewing pinned events, thanks to @istalri:federated.nexus for this!

Reviewer 2: ```suggestion
> - Support for viewing pinned events, thanks to [@istalri:federated.nexus](https://matrix.to/#/@istalri:federated.nexus) for this!
```

>     - **`hmac-sha256` is gone for SAS verification.** Only `hkdf-hmac-sha256` is offered as a message authentication code, so verifying against clients that only support `hmac-sha256` is no longer possible.
>     - **`Account.remove_one_time_keys()` has been removed**, as it is no longer supported by the underlying library.
> - **Dropped Python 3.8 and 3.9.** The minimum supported version is now **Python 3.10**. **Python 3.14 is now supported.** Supported: 3.10, 3.11, 3.12, 3.13, 3.14.
>
> #### Additions
>
> - Added `unread_thread_notifications` to `SyncResponse`.
>
> #### Bug fixes
>
> - Fixed `get_openid_token`, which requires an empty JSON body to be sent.
> - Fixed printing of `FileResponse` when a download is saved to a file.
>
> #### Internal / maintenance
>
> - Adopted [`uv`](https://docs.astral.sh/uv/) for project management, applied `pyupgrade` and linting, and bumped dependencies.
> - Unpinned dependencies for more flexible version resolution.
> - Upgraded the underlying `vodozemac` to **0.10.0**.

> ### vodozemac ([website](https://github.com/matrix-org/vodozemac))

Reviewer: this one is on me for categorising this as vodozemac when it is actually vodozemac-python. i also assumed it would pop up in the crypto section or so. @MTRNord do we keep it as is?

Reviewer 2: I think its good enough. I am fine with either. There is a mention that its python later on too.

> An implementation of Olm and Megolm in pure Rust.

Reviewer: ```suggestion
### vodozemac-python ([website](https://github.com/matrix-nio/vodozemac-python))

Python bindings for vodozemac, the implementation of Olm and Megolm in pure Rust.
```

>
> #### Internal / maintenance
>
> - Upgraded `pyo3` 0.28 → **0.29** and `maturin` to **1.14**.
> - Bumped `thiserror` and refreshed transitive Rust and Python dev dependencies.
> - Session configuration continues to use protocol **version 1** for all Olm/Megolm sessions; this is unchanged from 0.9.

## Dept of Ops 🛠

### Matrix Connectivity Tester ([website](https://connectivity-tester.mtrnord.blog))

A web interface to debug and troubleshoot Matrix connectivity issues.

[MTRNord (they/them)](https://matrix.to/#/@mtrnord:mtrnord.blog) reports

> I just released v0.1.0 (and v0.1.1) of the elixir connectivity tester. This one's a bigger change than the last few: the Rust backend and the separate frontend are now merged into a single Elixir/Phoenix app that ships the web UI, API, alerting, and an OAuth2/OIDC provider together in one deployable unit. Development also moved off GitHub, so the old two-repo setup is gone.
>
> #### What's Changed
>
> - Unified the previously separate backend (Rust) and frontend into one Elixir/Phoenix app. All existing checks (federation, DNS/well-known, TLS, split-brain detection, MatrixRTC, the OAuth2-backed alerts) carried over

> > - New federation checks along the way: an outgoing-federation self-check (does the target server actually reach us back over federation, not just the other way round) (thanks @networkexception:nwex.de for that idea!), dual-stack/protocol asymmetry detection, and signing-key-expiry warnings

Reviewer: ```suggestion
> - New federation checks along the way: an outgoing-federation self-check (does the target server actually reach us back over federation, not just the other way round) (thanks [@networkexception:nwex.de](https://matrix.to/#/@networkexception:nwex.de) for that idea!), dual-stack/protocol asymmetry detection, and signing-key-expiry warnings
```
